### PR TITLE
Removed buildtest build  option --disable-executor-check 

### LINKED ISSUE #1500: Remove option buildtest build --disable-executor-check
@Mariamajib i am assigning this task to you

# TODO 

- [ ] Remove option from argparse file https://github.com/buildtesters/buildtest/blob/devel/buildtest/cli/__init__.py
- [ ] Remove option from [bash completion](https://github.com/buildtesters/buildtest/blob/devel/bash_completion.sh) script 
- [ ] Remove logic for validation in config.py https://github.com/buildtesters/buildtest/blob/1f63db43ef2b10f40e6f18ece3ae376f5798a74e/buildtest/config.py#L122-L151
- [ ] Remove logic in main.py https://github.com/buildtesters/buildtest/blob/1f63db43ef2b10f40e6f18ece3ae376f5798a74e/buildtest/main.py#L125-L128 and remove variable in method invocation https://github.com/buildtesters/buildtest/blob/1f63db43ef2b10f40e6f18ece3ae376f5798a74e/buildtest/main.py#L135

### PR DESCRIPTION
**Pull request by Mariamajib**

### CODE REVIEW

File: buildtest/cli/__init__.py [hunk 1]
```diff
@@ -670,7 +670,6 @@ def build_menu(subparsers):
         type=positive_number,
     )
     extra_group.add_argument(
```
Comment (szuananwar) [inline buildtest/cli/__init__.py:670]: Please remove lines from 672-676 as we discussed.

File: bash_completion.sh [hunk 2]
```diff
@@ -162,7 +162,7 @@ _buildtest ()
   #case "${prev}" in
     build|bd)
       local shortoption="-b -e -et -f -m -s -t -u -x -xt"
-      local longoption="--buildspec --disable-executor-check --executor --executor-type --exclude --exclude-tags --filter --helpfilter --limit --maxpendtime --modules --module-purge --nodes --pollinterval --procs --profile --rerun --remove-stagedir --retry --save-profile --stage --tags --timeout --unload-modules"
+      local longoption="--buildspec --executor --executor-type --exclude --exclude-tags --filter --helpfilter --limit --maxpendtime --modules --module-purge --nodes --pollinterval --procs --profile --rerun --remove-stagedir --retry --save-profile --stage --tags --timeout --unload-modules"
```
Comment (szuananwar) [inline bash_completion.sh:165]: This is correct, no more deletion needed.

File: buildtest/config.py
Comment (szuananwar) [inline buildtest/config.py:1]: This is correct, no more deletion needed.

File: buildtest/main.py
Comment (szuananwar) [inline buildtest/main.py:1]: This is correct, no more deletion needed.

**Review by szuananwar — COMMENTED** [review]
@Mariamajib Please see my comments.

**Comment by szuananwar** [comment]
We will need to work on documenting these changes partner @Mariamajib! 
@shahzebsiddiqui right?

**Comment by shahzebsiddiqui** [comment]
No this doesn't need to be documented we are removing this feature

**Comment by szuananwar** [comment]
@shahzebsiddiqui I meant remove it from the document  too: https://buildtest.readthedocs.io/en/devel/command.html#extra

File: buildtest/config.py [hunk 3]
```diff
@@ -119,7 +119,7 @@ def detect_system(self):
 
         # self.localexecutors = list(self.target_config["executors"]["local"].keys())
 
-    def validate(self, validate_executors=True, moduletool=None):
+    def validate(self, moduletool=None):
```
Comment (shahzebsiddiqui) [inline buildtest/config.py:122]: can you also update the docstring there is line below that has 

```
validate_executors (bool): Check executor settings. This is the default behavior but can be disabled
```

**Review by shahzebsiddiqui — CHANGES_REQUESTED** [review]
please make this one change you should be all set

**Comment by shahzebsiddiqui** [comment]
> @shahzebsiddiqui I meant remove it from the document too: https://buildtest.readthedocs.io/en/devel/command.html#extra

This is auto-generated so it will be removed, you could probably see it in the generated docs but right now they are not passing

**Comment by shahzebsiddiqui** [comment]
@Mariamajib please fix the CI checks and documentation is not building

**Comment by Mariamajib** [comment]
@szuananwar  I have removed lines from 672-676 as stated

File: buildtest/config.py (hunk 3 above)
Comment (Mariamajib) [reply buildtest/config.py:122]: > can you also update the docstring there is line below that has
> 
> ```
> validate_executors (bool): Check executor settings. This is the default behavior but can be disabled
> ```

@shahzebsiddiqui I have updated the docstring

File: buildtest/cli/__init__.py (hunk 1 above)
Comment (Mariamajib) [reply buildtest/cli/__init__.py:670]: @szuananwar I have removed lines from 672-676 as stated

**Comment by shahzebsiddiqui** [comment]
@Mariamajib i am seeing issues with regression test can you please take a look. Also documentation is not building

**Comment by shahzebsiddiqui** [comment]
I did a quick search https://github.com/search?q=repo%3Abuildtesters%2Fbuildtest%20validate_executors&type=code and found a few more references to where we set `validate_executors=True`. Please fix this

**Comment by Mariamajib** [comment]
@shahzebsiddiqui  Thank you. I will fix it!

**Pull request merged**